packages/analytics-core/src/plugins/destination.ts

```diff
@@ -117,6 +120,11 @@ export class Destination implements DestinationPlugin {
   }
 
   async flush(useRetry = false) {
+    // Skip flush if offline
+    if (this.config.offline) {
+      return;
```

Comment:
I feel we should log here to help with debugging. e.g. `config.logger.debug("Skipping flush while offline.")

Reply:
Good point! I added logging messages when goes to online and offline too in https://github.com/amplitude/Amplitude-TypeScript/pull/644/commits/c2c585304a4e6381c648fe471e9603cc5accd735